SentinelOne/sentinelone_module/logs/connector.py

```diff
@@ -0,0 +1,123 @@
+from functools import cached_property
+import time
+
+from datetime import datetime, timedelta, timezone
+from prometheus_client import Counter, Histogram
+from management.mgmtsdk_v2.exceptions import UnauthorizedException
+from management.mgmtsdk_v2.services.activity import ActivitiesFilter
+from management.mgmtsdk_v2.services.threat import ThreatQueryFilter
+from management.mgmtsdk_v2_1.mgmt import Management
+from sekoia_automation.connector import Connector
+
+from sentinelone_module.logs.configuration import SentinelOneLogsConnectorConfiguration
+
+SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE = 1
+
+# Declare prometheus metrics
+prom_namespace = "sicconfapi_intakes"
+
+OUTCOMING_EVENTS = Counter(
+    name="forwarded_events",
+    documentation="Number of events forwarded to SEKOIA.IO",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+FORWARD_EVENTS_DURATION = Histogram(
+    name="forward_events_duration",
+    documentation="Duration to collect and forward events",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+
+class SentinelOneLogsConnector(Connector):
+    configuration: SentinelOneLogsConnectorConfiguration
+
+    @cached_property
+    def management_client(self):
+        return Management(hostname=self.configuration.management_domain, api_token=self.configuration.api_key)
+
+    @property
+    def logs_retrieval_start_date(self) -> str:
+        return (
+            datetime.now(tz=timezone.utc) - timedelta(days=SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE)
+        ).isoformat()
+
+    def validate_settings(self):
+        if not self.configuration.api_key:
+            self.log(f"Connector {self.configuration.uuid} missing api_key value", level="warning")
+            return
+
+        if not self.configuration.management_domain:
+            self.log(f"Connector {self.configuration.uuid} missing management_domain value", level="warning")
+            return
+
+        try:
+            self.management_client.system.get_info()
+            self.configuration.validated = True
+            self.configuration.last_activity_created_at = self.logs_retrieval_start_date
+            self.configuration.last_threat_created_at = self.logs_retrieval_start_date
```

Comment:
Can you define [a context](https://github.com/SEKOIA-IO/automation-library/blob/main/Checkpoint/connectors/checkpoint_harmony_mobile.py#L45) and use it to define the starting date?
We use the context to store [the progression of the connector](https://github.com/SEKOIA-IO/automation-library/blob/main/Checkpoint/connectors/checkpoint_harmony_mobile.py#L99-L122) in the collect of the events.
However, as two threads are used in the connector to collect activities and threats, I suggest you taking inspiration from the [Duo connector](https://github.com/SEKOIA-IO/automation-library/blob/main/Duo/duo/connector.py), that also use threads to collect events.